ArduPlane/mode_guided.cpp

```diff
@@ -64,3 +64,34 @@ void ModeGuided::set_radius_and_direction(const float radius, const bool directi
     active_radius_m = constrain_int32(fabsf(radius), 0, UINT16_MAX);
     plane.loiter.direction = direction_is_ccw ? -1 : 1;
 }
+
+void ModeGuided::update_target_altitude()
+{
+#if OFFBOARD_GUIDED == ENABLED
+    if (!((plane.guided_state.target_alt_time_ms != 0) || (plane.guided_state.target_alt > -0.001))) {
+        Mode::update_target_altitude();
+        return;
+    }
+
+    // offboard altitude demanded
+    uint32_t now = AP_HAL::millis();
+    float delta = 1e-3f * (now - plane.guided_state.target_alt_time_ms);
+    plane.guided_state.target_alt_time_ms = now;
+    // determine delta accurately as a float
+    float delta_amt_f = delta * plane.guided_state.target_alt_accel;
+    // then scale x100 to match last_target_alt and convert to a signed int32_t as it may be negative
+    int32_t delta_amt_i = (int32_t)(100.0 * delta_amt_f); 
+    Location temp {};
+    temp.alt = plane.guided_state.last_target_alt + delta_amt_i; // ...to avoid floats here, 
+    if (is_positive(plane.guided_state.target_alt_accel)) {
+        temp.alt = MIN(plane.guided_state.target_alt, temp.alt);
+    } else {
+        temp.alt = MAX(plane.guided_state.target_alt, temp.alt);
+    }
+    plane.guided_state.last_target_alt = temp.alt;
+    plane.set_target_altitude_location(temp);
+    plane.altitude_error_cm = plane.calc_altitude_error_cm();
+#else
+    Mode::update_target_altitude();
+#endif
```

Comment:
Cant just not override in `mode.h` as including `config.h` to get `OFFBOARD_GUIDED` breaks the build in lots of exciting ways.